### ISSUE TITLE
Fix Will-O-Wisp Scoring

### ISSUE THREAD

**Issue by Pawkkie** [rh-hideout/pokeemerald-expansion#5272]
### Description

Grabbing this from @iriv24 on Discord so we don't lose it:

suggestion for AI behavior: only prefer will-o-wisp if the player's mon has a physical attacking move or a higher attack stat than special attack or something along those lines.
situation where i encnoutered this:
ai has flags: AI_FLAG_CHECK_BAD_MOVE | AI_FLAG_TRY_TO_FAINT | AI_FLAG_CHECK_VIABILITY | AI_FLAG_SMART_SWITCHING | AI_FLAG_OMNISCIENT | AI_FLAG_SMART_MON_CHOICES
AI mon:
```
{
    .iv = TRAINER_PARTY_IVS(31, 31, 31, 31, 31, 31),
    .lvl = 31,
    .species = SPECIES_RAPIDASH_GALARIAN,
    .nature = NATURE_JOLLY,
    .ability = ABILITY_IMMUNITY,
    .heldItem = ITEM_SITRUS_BERRY,
    .moves = {MOVE_FLAME_WHEEL, MOVE_WILL_O_WISP, MOVE_COVET, MOVE_QUICK_ATTACK}
    },
```
Player mon: level 32 modest toedscuel with mega drain, mud shot, hex, stun spore. 
Rapidash went for will-o-wisp when i dont have any physical attack moves and have shit attack anyway (47 attack vs 79 special attack), i would expect flame wheel to be the optimal play in this scenario

### Version

1.9.1 (Latest release)

### Discord contact info

@iriv24

**Comment by Pawkkie** [comment]
Update on addressing this, it appears it may not be an issue. The following test representing the scenario passes on current upcoming, current master, 1.9.1, 1.9.0, and the project iriv was using when he encountered the behaviour. Here's the test we used:

```
AI_SINGLE_BATTLE_TEST("Iriv Test")
{
    GIVEN {
        AI_FLAGS(AI_FLAG_CHECK_BAD_MOVE | AI_FLAG_CHECK_VIABILITY | AI_FLAG_TRY_TO_FAINT | AI_FLAG_SMART_MON_CHOICES | AI_FLAG_SMART_SWITCHING | AI_FLAG_OMNISCIENT);
        PLAYER(SPECIES_TOEDSCRUEL) { Level(32); Ability(ABILITY_MOLD_BREAKER); (ITEM_SITRUS_BERRY); Nature(NATURE_MODEST); Moves(MOVE_MEGA_DRAIN, MOVE_MUD_SHOT, MOVE_HEX, MOVE_STUN_SPORE); }
        OPPONENT(SPECIES_RAPIDASH_GALARIAN) { Level(31); Ability(ABILITY_IMMUNITY); (NATURE_JOLLY); Item(ITEM_SITRUS_BERRY); (MOVE_FLAME_WHEEL, MOVE_WILL_O_WISP, MOVE_COVET, MOVE_QUICK_ATTACK); }
    } WHEN {
        TURN { MOVE(player, MOVE_HEX); EXPECT_MOVE(opponent, MOVE_FLAME_WHEEL); }
    }
}
```

Thankfully iriv still had a save state from right before the battle recorded (I wish all our bug reports were like this), and on further testing. found the following:

![pokeemerald-0](https://github.com/user-attachments/assets/f781654f-7477-49dd-9465-7e99f2bc0fe1)

Notes that weren't in the initial report but have been added, tested, and still pass:
- Toedscruel has Mold Breaker and a Sitrus Berry
- Rapidash Galar is Fire/Fairy instead of Fairy/Psychic

I then recreated the battle on a clean expansion 1.9.0 with these modifications, where the test still passes, and got the same move-tying result. It turns out the test passes with both MOVE_FLAME_WHEEL and MOVE_WILL_O_WISP.

Following up further in discord, will update this issue again when further progress is made :)

**Comment by Pawkkie** [comment]
Fixed by #5356
